File: src/resources/identity.ts
import type { HttpClient } from '../http.js';
import type { DeveloperIdentity, OrgIdentity, AgentIdentity, SignupOpts, RegisterAgentOpts } from '../types/identity.js';

export class IdentityResource {
  constructor(private http: HttpClient) {}

  async signup(opts: SignupOpts): Promise<DeveloperIdentity> {
    return this.http.post<DeveloperIdentity>('/v1/identity/signup', opts);
  }

  async createOrg(opts: { name: string }): Promise<OrgIdentity> {
    return this.http.post<OrgIdentity>('/v1/identity/orgs', opts);
  }

  async registerAgent(owner: string, opts: RegisterAgentOpts): Promise<AgentIdentity> {
    return this.http.post<AgentIdentity>(`/v1/identity/agents/${owner}`, opts);
  }

  async lookup(owner: string, name: string): Promise<AgentIdentity> {
    return this.http.get<AgentIdentity>(`/v1/identity/agents/${owner}/${name}`);
  }
}

File: packages/python/invariance/resources/identity.py
from __future__ import annotations

from typing import Any

from ..http_client import HttpClient
from ..types import DeveloperIdentity, OrgIdentity, AgentIdentity, SignupOpts, RegisterAgentOpts


class IdentityResource:
    def __init__(self, http: HttpClient) -> None:
        self._http = http

    async def signup(self, opts: SignupOpts) -> DeveloperIdentity:
        return await self._http.post("/v1/identity/signup", opts)

    async def create_org(self, opts: dict[str, str]) -> OrgIdentity:
        return await self._http.post("/v1/identity/orgs", opts)

    async def register_agent(self, owner: str, opts: RegisterAgentOpts) -> AgentIdentity:
        return await self._http.post(f"/v1/identity/agents/{owner}", opts)

    async def lookup(self, owner: str, name: str) -> AgentIdentity:
        return await self._http.get(f"/v1/identity/agents/{owner}/{name}")
